pvanalytics/system.py

```diff
@@ -1 +1,119 @@
 """Functions for identifying system characteristics."""
+import pandas as pd
+import numpy as np
+import pvlib
+from pvanalytics.util import _fit, _group
+
+
+def _peak_times(data):
+    minute_of_day = pd.Series(
+        data.index.hour * 60 + data.index.minute,
+        index=data.index
+    )
+    peak_minutes = _group.by_day(data).apply(
+        lambda day: pd.Timedelta(
+            minutes=_fit.quadratic_idxmax(
+                x=minute_of_day[day.index],
+                y=day,
+                model_range=range(0, 1440)
+            )
+        )
+    )
+    return pd.DatetimeIndex(
+        np.unique(data.index.date),
+        tz=data.index.tz
+    ) + peak_minutes
+
+
+def infer_orientation_least_squares(power_or_poa, daytime, tilts,
+                                    azimuths, solar_azimuth,
+                                    solar_zenith, ghi, dhi, dni):
+    """Determine system azimuth and tilt from power or POA using least
+    squares.
+
+    Solar noon is estimated on each day by fitting a quadratic to data
+    in `power_or_poa` and finding the vertex of the fit. A brute force
+    search is performed on clearsky POA irradiance for all pairs of
+    candidate azimuths and tilts (`azimuths` and `tilts`) to find the
+    pair that results in the closest azimuth at solar noon to the
+    azimuths calculated from the curve fitting step. Closest is
+    determined by minimizing the sum of squared error between the
+    solar azimuth at solar noon on each day in `power_or_poa` and the
+    solar azimuth at maximum clearsky POA irradiance.
+
+    The accuracy of the tilt and azimuth returned by this function will
+    vary with the time-resolution of the clearsky and solar position
+    data. For the best accuracy pass `solar_azimuth`, `solar_zenith`,
+    and the clearsky data (`ghi`, `dhi`, and `dni`) with one-minute
+    timestamp spacing. If `solar_azimuth` has timestamp spacing less
+    than one minute it will be resampled and interpolated to estimate
+    azimuth at each minute of the day. Regardless of the timestamp
+    spacing these parameters must cover the same days as
+    `power_or_poa`.
+
+    Parameters
+    ----------
+    power_or_poa : Series
+        Timezone localized series of power or POA irradiance
+        measurements.
+    daytime : Series
+        Boolean series with True for values when it is daytime.
```

Comment:
Could this be inferred from `solar_zenith` with a zenith limit? Maybe that has been tried.

Reply:
I was a little confused by this parameter after your last review so I went back to my notes. It was originally called 'sunny' because it was supposed to mask out cloudy days as well as night. I may have gotten a little overzealous in my renaming of parameters from the original PVFleets implementation. Should probably go back and rename this param `clearsky` or something along those lines.

Reply:
Aha, this algorithm makes a lot more sense if `power_or_poa` has already been filtered to times with clear-sky conditions, or `sunny` is a filter that subsets `power_or_poa` to clear-sky conditions.

Reply:
I'll change the parameter name back to `sunny` or `clearsky` and write a better description.